### PR TITLE
[Compute] `az vm/vmss create`: Fix help message for `--public-ip-address` parameter

### PR DESCRIPTION
**Pull request by yanzhudd**
**Related command**

`az vm/vmss create`

**Description**

Fix help message: https://github.com/Azure/azure-cli/pull/30363#discussion_r1843266003
In PowerShell, the empty string should be '""' instead of '' or "":
image

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

**Comment by zhoxing-ms** [comment]
> In PowerShell, the empty string should be '""' instead of '' or ""

Please add a screenshot of the repaired effect in PowerShell to the PR description

**Comment by zhoxing-ms** [comment]
@bebound Please help review this PR as well

**Comment by yanzhudd** [comment]
> > In PowerShell, the empty string should be '""' instead of '' or ""
> 
> Please add a screenshot of the repaired effect in PowerShell to the PR description

updated to description section

**Review by zhoxing-ms — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/vm/_params.py [hunk 1]
```diff
@@ -1136,7 +1136,7 @@ def load_arguments(self, _):
...
             c.argument('nics', nargs='+', help='Names or IDs of existing NICs to attach to the VM. The first NIC will be designated as primary. If omitted, a new NIC will be created. If an existing NIC is specified, do not specify subnet, VNet, public IP or NSG.')
             c.argument('private_ip_address', help='Static private IP address (e.g. 10.0.0.5).')
-            c.argument('public_ip_address', help='Name of the public IP address when creating one (default) or referencing an existing one. Can also reference an existing public IP by ID or specify "" for None (\'""\' in Azure CLI using PowerShell or --% operator). For Azure CLI using powershell core edition 7.3.4, specify '' or "" (--public-ip-address '' or --public-ip-address "")')
+            c.argument('public_ip_address', help='Name of the public IP address when creating one (default) or referencing an existing one. Can also reference an existing public IP by ID or specify "" for None (\'""\' in Azure CLI using PowerShell or --% operator). For Azure CLI using PowerShell core edition 7.3.4, specify \'""\' (--public-ip-address \'""\')')
```
Comment (bebound) [inline src/azure-cli/azure/cli/command_modules/vm/_params.py:1139]: Not sure why `For Azure CLI using PowerShell core edition 7.3.4, xxxx` is introduced. It's redundant and can be omitted as it repeats the previous sentence.

Comment (yanzhudd) [reply src/azure-cli/azure/cli/command_modules/vm/_params.py:1139]: The reason why I didn't remove the `For Azure CLI using PowerShell core edition 7.3.4, xxxx` sentence is that I'm not sure about the behavior of versions prior to `7.3.4`. The `--% operator` doesn't seem to work for `7.3.4`, but not sure if it works for the previous versions.

**Review by bebound — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/vm/_params.py [hunk 2]
```diff
@@ -1136,7 +1136,7 @@ def load_arguments(self, _):
             c.argument('subnet_address_prefix', help='The subnet IP address prefix to use when creating a new VNet in CIDR format.')
             c.argument('nics', nargs='+', help='Names or IDs of existing NICs to attach to the VM. The first NIC will be designated as primary. If omitted, a new NIC will be created. If an existing NIC is specified, do not specify subnet, VNet, public IP or NSG.')
             c.argument('private_ip_address', help='Static private IP address (e.g. 10.0.0.5).')
-            c.argument('public_ip_address', help='Name of the public IP address when creating one (default) or referencing an existing one. Can also reference an existing public IP by ID or specify "" for None (\'""\' in Azure CLI using PowerShell or --% operator). For Azure CLI using powershell core edition 7.3.4, specify '' or "" (--public-ip-address '' or --public-ip-address "")')
+            c.argument('public_ip_address', help='Name of the public IP address when creating one (default) or referencing an existing one. Can also reference an existing public IP by ID or specify "" or \'\' for None (\'""\' in Azure CLI using PowerShell).')
```
Comment (bebound) [inline src/azure-cli/azure/cli/command_modules/vm/_params.py:1139]: This is much more concise now.

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**